openhands/linter/linter.py

```diff
@@ -33,3 +34,33 @@ def lint(self, file_path: str) -> list[LintResult]:
             if res:
                 return res
         return []
+
+    def lint_file_diff(
+        self, original_file_path: str, updated_file_path: str
+    ) -> list[LintResult]:
+        """Only return lint errors that are introduced by the diff.
+
+        Args:
+            original_file_path: The original file path.
+            updated_file_path: The updated file path.
+
+        Returns:
+            A list of lint errors that are introduced by the diff.
+        """
+        updated_lint_error: list[LintResult] = self.lint(updated_file_path)
+
+        with open(original_file_path, 'r') as f:
+            original_file_content = f.read()
+        with open(updated_file_path, 'r') as f:
+            updated_file_content = f.read()
+        diff = get_diff(original_file_content, updated_file_content)
+        changes = parse_diff(diff)
+
+        # Select errors that are introduced by the new diff
+        selected_errors = []
+        for change in changes:
+            new_lineno = change.new
+            for error in updated_lint_error:
+                if error.line == new_lineno:
```

Comment:
Is `new_lineno` guaranteed to be non-conflicting with old line numbers (before edit)?

Reply:
What do you mean by "non-conflicting"? the `new_lineno` is the changed lines in the new file - which should only relevant in the new version of the file?

Reply:
I move the linter related changes to a separate PR - will try to add more testcases from your branch :)
https://github.com/All-Hands-AI/OpenHands/pull/4409

Reply:
> What do you mean by "non-conflicting"?

Never mind, I thought you also linted twice in your approach. Now it seems you only lint once. Anyways let's discuss in your new PR.